spring-kafka/src/main/java/org/springframework/kafka/listener/CompositeRecordInterceptor.java

```diff
@@ -78,12 +78,13 @@ public void failure(ConsumerRecord<K, V> record, Exception exception, Consumer<K
 	}
 
 	@Override
-	public void beforePoll(Consumer<K, V> consumer) {
-		this.delegates.forEach(del -> del.beforePoll(consumer));
+	public void setupThreadState(Consumer<?, ?> consumer) {
+		this.delegates.forEach(del -> del.setupThreadState(consumer));
 	}
 
 	@Override
-	public void clearThreadState(Consumer<K, V> consumer) {
+	public void clearThreadState(Consumer<?, ?> consumer) {
```

Comment:
Should it also delegate `setupThreadState()` and `clearThreadState()` inherited from `ThreadStateProcessor` and the new `afterRecord()` method?